---
title: "mTOR in TCGA studies"
subtitle: "preliminary exploration"
date: "Figures last updated: `r Sys.time()`"
output: 
  flexdashboard::flex_dashboard:
    orientation: rows
    source_code: embed
    theme: lumen
params:
  maf_dir: "../GDCdata/harmonized"
  res_dir: "../results/harmonized"
  names_tr: "../ref_files/cancer_types.txt"
---

```{r global, include=FALSE}
library(maftools)
library(dplyr)
library(ggplot2)
library(plotly)
library(kableExtra)
source("gsea.R", local = knitr::knit_global())
knitr::opts_chunk$set(echo = F, warning = F, message = F, fig.align = 'center', out.width = '100%')
```

```{css, echo=FALSE}

.slick-arrow.slick-prev {
  left: 100px;
  z-index: 1;
}
.slick-arrow.slick-next {
  right: 100px;
}

```

Inputs {.sidebar}
=======================================================================
### Analysis info

Ran [MutSigCV](https://software.broadinstitute.org/cancer/cga/mutsig) to get p-values. These indicate if the mutation frequency of a gene is significant, accounting for the background mutation rate of the patient, the gene expression level, gene replication timing, and chromatin state.

Data retrieved from [GDC data portal](https://portal.gdc.cancer.gov/). Mutation displayed is from MC3 data project, and aligned to hg19. MAF files an oncoplots were prepared using [maftools](https://www.bioconductor.org/packages/release/bioc/html/maftools.html). Expression data is from HTseq counts experiemnts, aligned to hg38. 

*** 

**Gene Significance**: MutSigCV p-values are plotted for each available TCGA cancer type, for a select set of genes. 

**Oncoplots**: show the frequency and nature of mutations in the patients of a given cancer type. Gene sets are boxed in grey. Barplot on right illustrates p-value for a given gene, as determined with MutSigCV. 

**Waterfall plots**: show co-occuring alterations in a pathway for each patient. Blue are oncogenes, red are tumour suppressors. 

**Volcano plots** show log2 fold change in RNA-seq expression counts for PI3K/PTEN vs MTOR alterated cases. 

**Kegg patway plots** show log2 fols change in RNA-seq expression counts in context of the signalling pathway (protein view). Blue is downregulated relative to reference class, and red is upregulated. 


Gene Significance 
=======================================================================

```{r, manifest, include = F, eval = T}
# mafs
maf_dir <- params$maf_dir
maf_fns <- list.files(maf_dir, pattern = "*.maf")
mafs <- sapply(paste0(maf_dir, "/", maf_fns), read.maf)
names(mafs) <- sapply(maf_fns, function(x){sub('\\.mutSig\\.maf$', '', x) })

# cancer types
cancer_names <- read.table(params$names_tr, header = T, sep = "\t")

# mutsigCV results
res_dir <- params$res_dir
res_fns <- list.files(res_dir, pattern = "*.sig_genes.txt")
goi <- c("TTN", "ATM", "WNT1", "KRAS", "TP53", "BRCA1", "BRCA2", "AKT1", "AKT2", "AKT3", "TSC1", "TSC2", "EGFR", 
         "FGFR1", "ERBB2", "ERBB3", "ERBB4", "ROS1", "MET", "ALK", "FLT1", "PDGFRA", "FLT3", "FLT4", "RET", "FGFR2", "FGFR3", "DEPDC5", "NPRL2", "NPRL3", "MIOS", "SEH1L", "SEC13", "WDR24", "WDR59", "SLC38A9", "PIK3CA", "PIK3CB", "PIK3CG", "PIK3CD", "PTEN", "MTOR")
mutsig_p <- data.frame(row.names = goi)

# set up mutsig dataframe for genes of interest
for (c in names(mafs)) {
  ord <- match(c, substr(res_fns, 1, nchar(res_fns)-14))
  mutsig_res <- read.table(paste0(res_dir, "/", res_fns[ord]), header=T)
  poi <- mutsig_res[mutsig_res$gene %in% goi, c(1,14)]
  mutsig_p[[c]] <- poi$p[match(goi, poi$gene)]
}
```

```{r sig-genes, eval = T}
df <- as.data.frame(t(mutsig_p))
sampleSizes <- sapply(mafs, function(x){as.numeric(x@summary[ID %in% 'Samples', summary])})
mutRates <- sapply(mafs, function(x){getGeneSummary(x)[match(goi, getGeneSummary(x)$Hugo_Symbol), MutatedSamples]})
mutRates <- as.data.frame(apply(mutRates, denom = sampleSizes, FUN = function(x, denom){ x / denom * 100}, MAR = 1))
colnames(mutRates) <- goi
df$cancer_type <- cancer_names$brief[match(rownames(df), cancer_names$abbrev)]

p <- plot_ly(data = df, type = 'scatter', mode = 'markers') 

for (g in names(df)) {
  if(g=='cancer_type'){next}
  p <- p %>% add_trace(x = ~ cancer_type, y = as.formula(paste0("~`", g, "`")),
                       name = g, 
                       text = paste("Cohort mutation rate: ", round(mutRates[[g]], digits = 2) , "%")
                       #color = mutRates[[g]]
                      )
}

p <- p %>% 
     add_trace(x = ~cancer_type, y = 0.1, type = "scatter", mode = "lines", name = "*",
               line = list(dash = "dash", width = 0.5, color = "black")) %>%
     add_trace(x = ~cancer_type, y = 0.05, type = "scatter", mode = "lines", name = "**",
               line = list(dash = "dash", width = 0.5, color = "black")) %>%
     layout(yaxis = list(title = "p-value"), xaxis = list(type = "category", title = ""),
            margin = list(b = 160, t = 70)) %>%
     plotly::config(toImageButtonOptions = list(format = "png", scale = 2))
#htmlwidgets::saveWidget(p, file = "img.html")

p
```

```{r pthw-assignment}

gene <- list(c("EGFR", "FGFR1", "ERBB2", "ERBB3", "ERBB4", "ROS1", "MET", "ALK", "FLT1", 
               "PDGFRA", "FLT3", "FLT4", "RET", "FGFR2", "FGFR3"),
             c("PIK3CA", "PIK3CB", "PIK3CG", "PIK3CD", "PTEN", "AKT3", "MTOR", "AKT2", "TSC2", "TSC1", "AKT1"),
             c("C7orf60", "GATSL1", "GATSL3"),
             c("DEPDC5", "NPRL2", "NPRL3"),
             c("MIOS", "SEH1L", "SEC13", "WDR24", "WDR59", "SLC38A9"),
             c("C12orf66", "ITFG2", "KPTN", "KIAA0467", "SESN2")
            )
pw_name <- c("RTKS pthw", "PI3K pthw", "SAMTOR", "GATOR1", "GATOR2", "KICSTOR")
pathways <- data.frame(unlist(gene), rep(pw_name, times = sapply(gene, length)))
pidx <- length(gene[[1]]) + length(gene[[2]]) + 1

```


KIRC
=======================================================================

```{r kirc, eval = T}
proj_name <- "TCGA-KIRC"
dea <- prepDE(proj_name, setdiff = F, count_thresh = 50, maf_fn = sprintf("../GDCdata/MC3/%s.mutSig.maf", proj_name), 
                   counts_fn = sprintf("../GDCdata/HTseq/%s_counts.rda", proj_name))
```

Row
-----------------------------------------------------------------------

### Volcano plot

```{r kirc-volcano}
volcano(dea$dds, dea$res, "PI3K/PTEN mutated vs MTOR mutated")
```

### PI3K waterfall

```{r kirc-waterfall}
PlotOncogenicPathways(dea$maf, pathways = "PI3K")
```

Row
-----------------------------------------------------------------------

### Kegg pathway view

```{r kirc-kegg}


kegg_highlight(dea$counts, dea$res)
```

### Oncoplot 

```{r kirc-onco, eval = F}
oncoplot(maf = mafs[[proj_name]], pathways = pathways, 
                  rightBarData = rbd, rightBarLims = c(0, 0.1), 
                  keepGeneOrder = T, gene_mar = 8, titleText = c)
```



UCEC
=======================================================================

DE analysis is restricted to MSI subtype only.

```{r ucec}
proj_name <- "TCGA-UCEC"
dea <- prepDE(proj_name, setdiff = F, count_thresh = 100, maf_fn = sprintf("../GDCdata/MC3-subtyped/%s.UCEC.MSI.mutSig.maf", proj_name), 
                   counts_fn = sprintf("../GDCdata/HTseq/%s_counts.rda", proj_name))
```

Row
-----------------------------------------------------------------------

### Volcano plot

```{r ucec-volcano}
volcano(dea$dds, dea$res, "PI3K/PTEN mutated vs MTOR mutated")
```

### PI3K waterfall

```{r ucec-waterfall}
PlotOncogenicPathways(dea$maf, pathways = "PI3K")
```

Row
-----------------------------------------------------------------------

### Kegg pathway view

```{r ucec-kegg}
kegg_highlight(dea$counts, dea$res)
```

### Oncoplot 

```{r ucec-onco, eval = F}
plot(1:20, col = 1:20)
```


COAD
=======================================================================

DE analysis is restricted to MSI subtype only.

```{r coad}
proj_name <- "TCGA-COAD"
dea <- prepDE(proj_name, setdiff = F, count_thresh = 50, maf_fn = sprintf("../GDCdata/MC3-subtyped/%s.GI.MSI.mutSig.maf", proj_name), 
                   counts_fn = sprintf("../GDCdata/HTseq/%s_counts.rda", proj_name))
```

Row
-----------------------------------------------------------------------

### Volcano plot

```{r coad-volcano}
volcano(dea$dds, dea$res, "PI3K/PTEN mutated vs MTOR mutated")
```

### PI3K waterfall

```{r coad-waterfall}
PlotOncogenicPathways(dea$maf, pathways = "PI3K")
```

Row
-----------------------------------------------------------------------

### Kegg pathway view

```{r coad-kegg}
kegg_highlight(dea$counts, dea$res)
```

### Oncoplot 

```{r coad-onco, eval = F}
plot(1:20, col = 1:20)
```

